frontend/src/components/BreadcrumbNavigation.js

```diff
@@ -5,64 +5,70 @@ import { useNavigation } from './NavigationContext';
-        if (regionId === null || regionId === 0 ) {
-            hasSubregions = true;
+  useEffect(() => {
+    const fetchAndSetAncestors = async () => {
+      try {
+        if (selectedRegion.id !== null && selectedRegion.id !== 0) {
+          const ancestors = await fetchAncestors(selectedRegion.id, selectedHierarchy.hierarchyId);
+          const reversedAncestors = ancestors.reverse();
+          setBreadcrumbItems([{id: 0, name: 'World', hasSubregions: true }, ...reversedAncestors]);
         } else {
-            try {
-                const region = await fetchRegion(regionId, selectedHierarchy.hierarchyId);
-                hasSubregions = region.hasSubregions;
-            } catch (error) {
-                console.error(`Error fetching region ${regionId}, consider the region as not having subregions:`, error);
-                hasSubregions = false;
-            }
+          setBreadcrumbItems([{ id: null, name: 'World', hasSubregions: true }]);
         }
-        setSelectedRegion({
-            id: regionId,
-            name: breadcrumbItems[index].name,
-            hasSubregions: hasSubregions,
-        });
-        // Truncate the breadcrumbItems array up to the clicked index + 1
-        setBreadcrumbItems(prevItems => prevItems.slice(0, index + 1));
-    };
+        setError(null);
+      } catch (error) {
+        console.error('Error fetching ancestors: ', error);
+        setError('An error occurred while fetching ancestors.');
+      }
+      };
+    fetchAndSetAncestors();
+  }, [selectedRegion, selectedHierarchy]);
+    
+    
+  const handleBreadcrumbClick = async (regionId, index) => {
+    try {
+      let hasSubregions;
+      if (regionId === null || regionId === 0 ) {
+        hasSubregions = true;
+      } else {
+        const region = await fetchRegion(regionId, selectedHierarchy.hierarchyId);
+        hasSubregions = region.hasSubregions;
+      }
+      setSelectedRegion({
+        id: regionId,
+        name: breadcrumbItems[index].name,
+        hasSubregions: hasSubregions,
+      });
+      // Truncate the breadcrumbItems array up to the clicked index + 1
+      setBreadcrumbItems(prevItems => prevItems.slice(0, index + 1));
+      setError(null);
+    } catch (error) {
+      console.error(`Error fetching region ${regionId}: `, error);
+      setError('An error occurred while fetching region details.');
+    }
+  };
 
-    return (
-        <Breadcrumbs aria-label="breadcrumb">
-            {breadcrumbItems.map((item, index) => (
-                <Typography
-                    color="inherit"
-                    key={index}
-                    onClick={() => handleBreadcrumbClick(item.id, index, item.hasSubregions)}
-                    style={{ cursor: 'pointer' }}
-                >
-                    {item.name}
-                </Typography>
-            ))}
-        </Breadcrumbs>
-    );
+  return (
+    <div>
+      {error && <div style={{ color: 'red' }}>{error}</div>}
```

Comment:
I suggest rendering the errors not per component but in some dedicated component (also see my comment about Provider). It would eliminate code duplication and make the error view more consistent.
I would suggest to render the component here:
https://github.com/OhmSpectator/track-your-regions/blob/359ffa40e14a7f27bb1b47f490d3b78c22916332/frontend/src/App.js#L14